mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,383 @@
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        title (str): title of the track
+        track_id (str): track id
+        humdrum_annotated_path (str): path to humdrum annotated score
+
+    Cached Properties:
+        keys (list): annotated local keys.
+        roman_numerals (list): annotated roman_numerals.
+        chords (list): annotated chords.
+        duration (int): relative duration
+        midi_path (str): path to midi
+        score (music21.stream.Score): music21 score
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+        self.humdrum_annotated_path = os.path.join(
+            self._data_home, self._track_paths["annotations"][0]
+        )
+        self.title = os.path.splitext(self._track_paths["annotations"][0])[0]
+
+    @core.cached_property
+    def score(self) -> music21.stream.Score:
+        return load_score(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def keys(self) -> Optional[KeyData]:
+        return load_key(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def keys_music21(self) -> Optional[List[dict]]:
+        return load_key_music21(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def roman_numerals(self) -> Optional[List[dict]]:
+        return load_roman_numerals(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def chords(self) -> Optional[ChordData]:
+        return load_chords(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def chords_music21(self) -> Optional[List[dict]]:
+        return load_chords_music21(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def duration(self) -> int:
+        return self.chords_music21[-1]["time"]
+
+    @core.cached_property
+    def midi_path(self) -> Optional[str]:
+        return load_midi_path(self.humdrum_annotated_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            metadata={
+                "duration": self.duration,
+                "title": self.title,
+                "key": self.keys,  # format is not the expected by keydata jams namespace
+                "chord": self.chords,  # format is not the expected by chorddata jams namespace
+                "keys_music21": self.keys_music21,
+                "chords_music21": self.chords_music21,
+                "roman_numerals": self.roman_numerals,
+                "midi_path": self.midi_path,
+                "humdrum_annotated_path": self.humdrum_annotated_path,
+            },
+        )
+
+
+def split_score_annotations(fhandle: TextIO):
```

Comment:
Can this function be hidden as well?
`_split_score_annotation`

Reply:
done